src/UiPath.Workflow/Activities/ScriptingJitCompiler.cs

```diff
@@ -88,6 +94,15 @@ public override LambdaExpression CompileExpression(ExpressionToCompile expressio
         return (LambdaExpression)results.ResultType.GetMethod("CreateExpression")!.Invoke(null, null);
     }
 
+    private Assembly CollectibleAlc_Resolving(AssemblyLoadContext loadContext, AssemblyName assemblyName)
```

Comment:
This all looks good. I'm curious how the AssemblyLoadContext knows enough to have the full AssemblyName but can't load the Assembly on its own.

Reply:
I must confess that it also seemed strange to me that even when I used a collectible AssemblyLoadContext, it was always searching for it using the full name at this stage, but for some reason this always lead to a standard `Assembly.Load` on that assembly (triggered by the `CreateExpression` call, as the expression used a collectible assembly value), which failed as this load context is collectible. 

Didn't really investigate this in-depth, but I suspect this may be a .NET 6 bug which may not repro in .NET 7 since I know that they rewrote some parts of the collectible assemblies handling.